### PR TITLE
Docs/update api urls

### LINKED ISSUE #2073: Docs: fix all links pointing to the API reference
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/medusajs/medusa/issues

### Issue Summary

Following our latest refactoring of the API reference, a bunch of links throughout the documentation must be fixed to point to the new API reference.

### How can this issue be resolved?

1. Find all occurrences of `https://docs.medusajs.com/api` under `docs/content`.
2. if the URL is old and does not work, replace it with the new working URL.

### Are you interested in working on this issue?

- [ ] I would like to fix this issue

### PR DESCRIPTION
**Pull request by Cwix**
This resolves #2073 

Side note: CONTRIBUTING.md says to create a PR against `develop` branch but I see the `master` and `develop` have diverged.

### CODE REVIEW

File: docs/content/quickstart/quick-start.md [hunk 1]
```diff
@@ -81,5 +81,5 @@ You can learn more about configuring your server and loading environment variabl
 
 - Install our [Next.js](../starters/nextjs-medusa-starter) or [Gatsby](../starters/gatsby-medusa-starter) storefronts to set up your ecommerce storefront quickly.
 - Install the [Medusa Admin](../admin/quickstart.md) to supercharge your ecommerce experience with easy access to configurations and features.
-- Check out the [API reference](https://docs.medusajs.com/api) to learn more about available endpoints available on your Medusa server.
+- Check out the [API reference](https://docs.medusajs.com/api/admin) to learn more about available endpoints available on your Medusa server.
```
Comment (shahednasser) [inline docs/content/quickstart/quick-start.md:84]:
```suggestion
- Check out the [API reference](https://docs.medusajs.com/api/store) to learn more about available endpoints available on your Medusa server.
```

**Review by shahednasser — CHANGES_REQUESTED** [review]
Thanks for your work @Cwix I just left comments for a few fixes below 😊

Also regarding your side note, documentation PRs can be created for the `master` branch.

File: docs/content/quickstart/quick-start.md (hunk 1 above)
Comment (Cwix) [reply docs/content/quickstart/quick-start.md:84]: I was thinking if admin is the right place for this 😄

**Review by shahednasser — APPROVED** [review]

**Comment by shahednasser** [comment]
Thanks! @Cwix 🚀

**Pull request merged**